vdr/src/contracts/cl/types/schema.rs

```diff
@@ -17,6 +17,55 @@ pub struct Schema {
     pub attr_names: Vec<String>,
 }
 
+impl Schema {
+    pub fn require_valid_id(&self, expected_schema_id: &SchemaId) -> VdrResult<()> {
+        let actual_schema_id = SchemaId::build(&self.issuer_id, &self.name, &self.version);
+
+        if expected_schema_id != actual_schema_id {
+            return Err(VdrError::InvalidSchema(actual_schema_id.to_string()));
+        }
+
+        Ok(())
+    }
+
+    fn require_name(&self) -> VdrResult<()> {
```

Comment:
Just put all checks inside of `validate` function. Do not see a big sense in defining separate functions.

Reply:
Moved checks